src/main/java/com/autotune/analyzer/recommendations/model/CostBasedRecommendationModel.java

```diff
@@ -216,9 +241,16 @@ private static double calculateMemoryUsage(IntervalResults intervalResults) {
         if (0 < numPods) {
             memUsage = (memUsageSum / numPods);
         }
-        memUsage = Math.max(memUsage, memUsageMax);
+        memUsageMax = Math.max(memUsage, memUsageMax);
+        memUsageMin = Collections.min(Arrays.asList(memUsage, memUsageMax, memUsageMin));
+        if (memUsageMin == 0.0)
+            memUsageMin = Collections.min(Arrays.asList(memUsage, memUsageMax));
```

Comment:
same here. What if memUsage is also zero ?

Reply:
Refactored the code with new logic, it takes care of both the CPU and memory cases.